be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -410,21 +417,22 @@ Status HdfsOrcScanner::do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk)
         return Status::EndOfFile("");
     }
 
-    size_t skip_num_values = 0;
-    Filter& filter = _chunk_filter;
     ChunkPtr& ck = *chunk;
-
     // this infinite for loop is for retry.
     for (;;) {
         orc::RowReader::ReadPosition position;
         size_t read_num_values = 0;
+        bool has_used_dict_filter = false;
         {
             SCOPED_RAW_TIMER(&_stats.column_read_ns);
             RETURN_IF_ERROR(_orc_adapter->read_next(&position));
             // read num values is how many rows actually read before doing dict filtering.
             read_num_values = position.num_values;
             RETURN_IF_ERROR(_orc_adapter->apply_dict_filter_eval_cache(_orc_row_reader_filter->_dict_filter_eval_cache,
-                                                                       &_chunk_filter));
+                                                                       &_dict_filter));
+            if (_orc_adapter->get_cvb_size() != read_num_values) {
```

Comment:
here add some comment, why _orc_adapter->get_cvb_size() != read_num_values, then has_used_dict_filter = true?

Reply:
That's right. Because dict filter has taken effect and filtered some rows.